var/spack/repos/builtin/packages/py-stevedore/package.py

```diff
@@ -12,9 +12,12 @@ class PyStevedore(PythonPackage):
     homepage = "https://docs.openstack.org/stevedore/latest/"
     pypi = "stevedore/stevedore-1.28.0.tar.gz"
 
+    version('3.5.0', sha256='f40253887d8712eaa2bb0ea3830374416736dc8ec0e22f5a65092c1174c44335')
     version('1.28.0', sha256='f1c7518e7b160336040fee272174f1f7b29a46febb3632502a8f2055f973d60b')
 
-    depends_on('python@2.6:')
+    depends_on('python@2.6:', type=('build', 'run'))
+    depends_on('python@3.6:', type=('build', 'run'), when='@3.5.0:')
 
-    depends_on('py-six@1.10.0:', type=('build', 'run'))
+    depends_on('py-six@1.10.0:', type=('build', 'run'), when='@:3.5')
     depends_on('py-pbr@2.0.0:2.1.0', type=('build', 'run'))
```

Comment:
```suggestion
    depends_on('py-pbr@2.0.0:2.1.0', type=('build', 'run'), when='@:3.4')
```
Otherwise this condition will be applied to all versions.